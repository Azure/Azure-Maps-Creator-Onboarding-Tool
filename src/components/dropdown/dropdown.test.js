import { render, screen, fireEvent } from '@testing-library/react';

import Dropdown from './index';

describe('Dropdown', () => {
  it('should render component', () => {
    const options = [
      {
        key: 'key1',
        text: 'text1',
      },
      {
        key: 'key2',
        text: 'text2',
      },
      {
        key: 'key3',
        text: 'text3',
      }
    ];
    const view = render(
      <Dropdown options={options} anotherAttr={1} someOtherAttr={'hello'}>A shrimp's heart is in its head.</Dropdown>
    );
    const dropdown = screen.getByRole('combobox');
    fireEvent.click(dropdown);
    expect(view).toMatchSnapshot();
    const optionsList = screen.getAllByRole('option');
    expect(optionsList.length).toBe(3);
    expect(optionsList[0]).toHaveTextContent('text1');
    expect(optionsList[1]).toHaveTextContent('text2');
    expect(optionsList[2]).toHaveTextContent('text3');
  });


  it('should render component with option groups', () => {
    const groups = [
      [
        {
          key: 'key1',
          text: 'text1',
        },
      ],
      [
        {
          key: 'key2',
          text: 'text2',
        },
        {
          key: 'key3',
          text: 'text3',
        }
      ],
    ];
    const view = render(
      <Dropdown optionGroups={groups} anotherAttr={1} someOtherAttr={'hello'}>A shrimp's heart is in its head.</Dropdown>
    );
    const dropdown = screen.getByRole('combobox');
    fireEvent.click(dropdown);
    expect(view).toMatchSnapshot();
    const optionsList = screen.getAllByRole('option');
    expect(optionsList.length).toBe(3);
    expect(optionsList[0]).toHaveTextContent('text1');
    expect(optionsList[1]).toHaveTextContent('text2');
    expect(optionsList[2]).toHaveTextContent('text3');
  });
});