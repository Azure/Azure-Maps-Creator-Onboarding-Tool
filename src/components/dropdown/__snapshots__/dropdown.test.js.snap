// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<body>
    <div>
      <div>
        <div
          aria-owns="fluent-listboxr0"
          class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
          style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem; width: 100%;"
        >
          <button
            aria-activedescendant="fluent-optionr1"
            aria-expanded="true"
            class="fui-Dropdown__button ___rl9y210_gr9ejb0 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl f19n0e5 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
            role="combobox"
            type="button"
          >
            <div
              class="emotion-0"
            >
              A shrimp's heart is in its head.
            </div>
            <span
              class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <span
            hidden=""
          />
        </div>
      </div>
    </div>
    <div
      class="___1uxlrft_djprso0 f1euv43f f15twtuk f1vgc2s3 f1e31b4d f494woh"
      dir="ltr"
    >
      <div
        class="fui-Listbox fui-Dropdown__listbox ___1te21m2_r0cuxf0 fxugw4r f1ewtqcl f22iagw f1vx9l62 f3hsy1e f5zp4f fpvhumw f1yog68k f13sgyd8 f1x4af0m f7x41pl fd55psn fruq291 fiut8dr f1hg901r f1aa9q02 f16jpd5f f1jar5jt fyu767a fmmk62d"
        id="fluent-listboxr0"
        role="listbox"
        style="position: fixed; left: 0px; top: 0px; margin: 0px;"
      >
        <div
          aria-selected="false"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-optionr1"
          role="option"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            text1
          </span>
        </div>
        <div
          aria-selected="false"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-optionr2"
          role="option"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            text2
          </span>
        </div>
        <div
          aria-selected="false"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-optionr3"
          role="option"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            text3
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<div>
    <div>
      <div
        aria-owns="fluent-listboxr0"
        class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
        style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem; width: 100%;"
      >
        <button
          aria-activedescendant="fluent-optionr1"
          aria-expanded="true"
          class="fui-Dropdown__button ___rl9y210_gr9ejb0 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl f19n0e5 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
          role="combobox"
          type="button"
        >
          <div
            class="emotion-0"
          >
            A shrimp's heart is in its head.
          </div>
          <span
            class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <span
          hidden=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dropdown should render component with option groups 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<body>
    <div>
      <div>
        <div
          aria-owns="fluent-listboxr4"
          class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
          style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem; width: 100%;"
        >
          <button
            aria-activedescendant="fluent-optionr6"
            aria-expanded="true"
            class="fui-Dropdown__button ___rl9y210_gr9ejb0 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl f19n0e5 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
            role="combobox"
            type="button"
          >
            <div
              class="emotion-0"
            >
              A shrimp's heart is in its head.
            </div>
            <span
              class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <span
            hidden=""
          />
        </div>
      </div>
    </div>
    <div
      class="___1uxlrft_djprso0 f1euv43f f15twtuk f1vgc2s3 f1e31b4d f494woh"
      dir="ltr"
    >
      <div
        class="fui-Listbox fui-Dropdown__listbox ___1te21m2_r0cuxf0 fxugw4r f1ewtqcl f22iagw f1vx9l62 f3hsy1e f5zp4f fpvhumw f1yog68k f13sgyd8 f1x4af0m f7x41pl fd55psn fruq291 fiut8dr f1hg901r f1aa9q02 f16jpd5f f1jar5jt fyu767a fmmk62d"
        id="fluent-listboxr4"
        role="listbox"
        style="position: fixed; left: 0px; top: 0px; margin: 0px;"
      >
        <div
          class="fui-OptionGroup ___17y07lh_1j2r6sh f22iagw f1vx9l62 fiut8dr f1xxzjds fu19d3i flylvvz f1ug5m11 f5642y ffdf81h flgyru6 f1fjgumo fwsdxdw f1sgo0dv"
          role="group"
        >
          <div
            aria-selected="false"
            class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
            id="fluent-optionr6"
            role="option"
            style="font-size: 0.813rem; color: black;"
          >
            <span
              aria-hidden="true"
              class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              text1
            </span>
          </div>
        </div>
        <div
          class="fui-OptionGroup ___17y07lh_1j2r6sh f22iagw f1vx9l62 fiut8dr f1xxzjds fu19d3i flylvvz f1ug5m11 f5642y ffdf81h flgyru6 f1fjgumo fwsdxdw f1sgo0dv"
          role="group"
        >
          <div
            aria-selected="false"
            class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
            id="fluent-optionr8"
            role="option"
            style="font-size: 0.813rem; color: black;"
          >
            <span
              aria-hidden="true"
              class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              text2
            </span>
          </div>
          <div
            aria-selected="false"
            class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
            id="fluent-optionr9"
            role="option"
            style="font-size: 0.813rem; color: black;"
          >
            <span
              aria-hidden="true"
              class="fui-Option__checkIcon ___1de1wj6_6yaxj50 fod5ikn f18b9hdq f1xk557c fd7fpy0 fvc9v3g"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.03 13.9 3.56 10a.75.75 0 0 0-1.12 1l4 4.5c.29.32.79.34 1.09.03l10.5-10.5a.75.75 0 0 0-1.06-1.06l-9.94 9.94Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              text3
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<div>
    <div>
      <div
        aria-owns="fluent-listboxr4"
        class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
        style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem; width: 100%;"
      >
        <button
          aria-activedescendant="fluent-optionr6"
          aria-expanded="true"
          class="fui-Dropdown__button ___rl9y210_gr9ejb0 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl f19n0e5 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
          role="combobox"
          type="button"
        >
          <div
            class="emotion-0"
          >
            A shrimp's heart is in its head.
          </div>
          <span
            class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <span
          hidden=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
